tests/queries/test_internal_trades.py

```diff
@@ -68,19 +80,26 @@ def token_slippage(
     return sum(a.amount for a in internal_trade_list if a.token == Address(token_str))
 
 
-class TestDuneAnalytics(unittest.TestCase):
-    def setUp(self):
-        self.dune = DuneAPI.new_from_environment()
+class TestInternalTrades(unittest.TestCase):
+    def setUp(self) -> None:
```

Comment:
Why do you choose to put the important dune connection elements into a testInternalTrades instance and store it there as part of a self? I would rather create a new object in: [tests/db/pg_client.py](https://github.com/cowprotocol/solver-rewards/pull/37/files#diff-53dcee04977aad01047b4959b6bda8d733cebc0a29b2f28486d34fb7b1606698) and then just instance this new object at the beginning of all test and later tear the object down. 
I think its nicer, if we have one object handling all the dune requests: building up the connection and executing the queries. This would make the whole connection store and handling more encapsulated and then could be easier shared in the package. Wdyt?

Reply:
This pattern is the way tests work in python. If you want there to be a DB connection inside this particular family of tests you need to create them in the `setUp`. I am not sure I understand your question here, but this means "these tests require a DB connection"

Reply:
I have written a new class `DuneRouter` inside of `pg_client` as I assume this is what you were referring to. Note that setup and teardown still require an instance of this connection.